src/pyprojroot/root.py

```diff
@@ -15,13 +15,11 @@
 )
 
 
-def as_start_path(start: Union[None, _PathType]) -> Path:
-    if start is None:
-        return Path.cwd()
-    if not isinstance(start, Path):
-        start = Path(start)
-    # TODO: consider `start = start.resolve()`
-    return start
+def as_start_path(start: Optional[_PathType]) -> Path:
+    """Convert path argument into normalised Path object."""
+    if start is not None:
+        return Path(start).expanduser().resolve()
+    return Path.cwd().resolve()
```

Comment:
I think `resolve` includes `expanduser` and `cwd` is always an absolute path. So these can be simplified.